browser/search_engines/search_engine_provider_service_factory.cc

```diff
@@ -47,7 +48,7 @@ KeyedService* InitializeSearchEngineProviderServiceIfNeeded(Profile* profile) {
     return new PrivateWindowSearchEngineProviderService(profile);
   }
 
-  return nullptr;
+  return new DefaultSearchEngineProviderService(profile);
```

Comment:
Hmm, using |profile| could make confusing.
So far, `SearchEngineProviderService` and it's subclasses are instantiated with otr profile because they are only used for otr profile. and all implementation assumes it's otr profile.
How about introducing new service instead of subclassing `SearchEngineProviderService`?

Reply:
Yeah, that's a good point, I was thinking about it, but decided that observing a single data point doesn't worth creating a whole new service :) If you feel like writing a separate one is more clear, I will do that

Reply:
actually, passing normal profile doesn't cause any side effect because you're only using `original_template_url_service_`. Or, how about adding some comments?

Reply:
Hmm, I think this service factory is not a proper place to get default search engine metric because this service was intended only for handling default search engine of non-normal profile.
but you only want normal profile's default search engine changing.
If we don't have yet, I think general-purpose service seems fine for logging metric if there is no good point for. WDYT?

Reply:
Or just using different service with this factory. is it strange?

Reply:
I think I'll just go for separate service&factory to reduce confusion

Reply:
fixed